src/TodoList.tsx

```diff
@@ -0,0 +1,217 @@
+/* eslint-disable no-lone-blocks */
+import { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+
+type Props = {
+  visibleTodos: Todo[];
+  deleteTodo: (todoId: number) => void;
+  deletingTodoId: number | null;
+  tempTodo: Todo | null;
+  isLoading: boolean;
+};
+
+export const TodoList: React.FC<Props> = ({
+  visibleTodos, deleteTodo, deletingTodoId, isLoading, tempTodo,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isEditing) {
+      inputRef.current?.focus();
+    }
+  }, [isEditing]);
+
+  const handleDoubleClick = () => {
+    setIsEditing(true);
+  };
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {/* This is a completed todo */}
+      {visibleTodos.map((todo) => (
+        <div
+          data-cy="Todo"
+          className={`todo ${todo.completed ? 'completed' : 'active'}`}
+          key={todo.id}
+        >
+          <label className="todo__status-label">
+            <input
+              data-cy="TodoStatus"
+              type="checkbox"
+              className="todo__status"
+              checked={todo.completed}
+            />
+          </label>
+
+          <span
+            data-cy="TodoTitle"
+            className="todo__title"
+            onDoubleClick={handleDoubleClick}
+          >
+            {todo.title}
+          </span>
+
+          {/* Remove button appears only on hover */}
+          <button
+            type="button"
+            key={todo.id}
+            className="todo__remove"
+            data-cy="TodoDelete"
```

Comment:
you can create the TodoItem component and move this code there, it will look better